mono/metadata/icall.c

```diff
@@ -153,84 +153,73 @@ mono_icall_get_file_path_prefix (const gchar *path)
 }
 #endif /* HOST_WIN32 */
 
-ICALL_EXPORT MonoObject *
-ves_icall_System_Array_GetValueImpl (MonoArray *arr, guint32 pos)
+ICALL_EXPORT MonoObjectHandle
+ves_icall_System_Array_GetValueImpl (MonoArrayHandle array, guint32 pos, MonoError *error)
 {
-	ERROR_DECL (error);
-	MonoClass *ac;
-	gint32 esize;
-	gpointer *ea;
-	MonoObject *result = NULL;
-
-	ac = (MonoClass *)arr->obj.vtable->klass;
+	MonoVTable *array_vtable = MONO_HANDLE_GETVAL (array, obj.vtable);
```

Comment:
`array_vtable` seems to only be used to get the domain and klass of `array`. So just use `mono_handle_class (array)` and `MONO_HANDLE_DOMAIN (array)`, respectively.